File: frontend/oei-web/src/app/domain/model/event/event-photo-consent.ts
// Explicit, per-event consent to publish a member's own photos on the event's live feed —
// distinct from (and never implied by) `EventRegistration`: signing up to attend an event never
// grants the right to publish that member's image (see the spec's "Consentement" section). Also
// distinct from third-party right-to-image, which stays the publishing member's own
// responsibility in V1 (no automated face-detection/consent-of-others check).
export interface EventPhotoConsent {
  readonly eventId: string;
  readonly memberId: string;
  readonly consentedAt: string;
}

export function createEventPhotoConsent(fields: EventPhotoConsent): EventPhotoConsent {
  return Object.freeze({ ...fields });
}

File: backend/domain/shared/src/main/java/global/oei/domain/shared/event/EventPhotoConsent.java
package global.oei.domain.shared.event;

import java.time.Instant;
import java.util.Objects;

import global.oei.domain.shared.member.MemberId;

/**
 * Explicit consent to publish a member's photos on an event's feed — distinct from
 * {@link EventRegistration}, which never implies photo-publication consent (see the
 * operation's own contract summary).
 */
public record EventPhotoConsent(String eventId, MemberId memberId, Instant consentedAt) {

    public EventPhotoConsent {
        Objects.requireNonNull(eventId, "eventId must not be null");
        Objects.requireNonNull(memberId, "memberId must not be null");
        Objects.requireNonNull(consentedAt, "consentedAt must not be null");
    }
}
